headers/headers.go

```diff
@@ -45,6 +45,11 @@ var (
 	ErrRequestNil = errors.New("error setting request header request was nil")
 )
 
+// IsNotFound return true if the err equals ErrHeaderNotFound return false otherwise
+func IsNotFound(err error) bool {
+	return err == ErrHeaderNotFound
```

Comment:
Would it be considered better practice to use https://golang.org/pkg/errors/#Is.  That way if the error is wrapped it will still return true?

Reply:
I've never seen that before - I am pretty sure we don't use that anywhere in any of our apps. Although now I question why not.

Reply:
I tried to implement this but couldn't find it anywhere even after updating to the latest version of things. After some digging it appears that `errors.Is()` is newly added in **GO 1.13** and not in **1.12**. 
I think we should use this going forward but it requires us to update to 1.13 first. For now I will leave my PR unchanged will the view of updating it when we officially migrate to 1.13